keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlVerificationRepository.kt

```diff
@@ -378,7 +378,10 @@ class SqlVerificationRepository(
         contexts // List<VerificationContext>
           // Creates a SELECT statement from each element of [contexts], where every column is a constant. e.g.:
           // SELECT 0, "staging", "myapp", "myapp-h123-v23.4" FROM dual
-          .mapIndexed { idx, v -> jooq.select(inline(idx), inline(v.environmentName), inline(v.artifactReference), inline(v.version)) as SelectOrderByStep<Record4<Int, String, String, String>> } // List<SelectOrderByStep<Record4<Int, String, String, String>>>
+          .mapIndexed { idx, v -> jooq.select(inline(idx).`as`(ind),
+                                              inline(v.environmentName).`as`(environmentName),
+                                              inline(v.artifactReference).`as`(artifactReference),
+                                              inline(v.version).`as`(artifactVersion)) as SelectOrderByStep<Record4<Int, String, String, String>> }
```

Comment:
Why was this failing? Because it would generate something like `SELECT 0, "test", "test", "myapp-h123-v23.4" FROM dual`… what's wrong with that?

Reply:
It's because of what happens when it's used in a subselect. This fails:

```
select * from (select 0, 'dev', 'dev', 'myartifact-1.0.0' from dual) x;
```

I have a documented example in #1821